compiler/rustc_target/src/spec/mod.rs

```diff
@@ -3456,6 +3754,7 @@ impl Target {
         key!(supports_xray, bool);
 
         base.update_from_cli();
+        base.check_consistency(/*for_json*/ true)?;
```

Comment:
given that you're adding a comment at both use sites, change it to an enum `ForJson::Yes`?